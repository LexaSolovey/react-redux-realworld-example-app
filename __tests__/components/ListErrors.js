import React from 'react';
import renderer from 'react-test-renderer';
import ListErrors from '../../src/components/ListErrors';

describe('>> ListErrors component', () => {
    it('renders null in case when errors is not exists', () => {
        const wrapper = renderer.create(<ListErrors />).toJSON();
        expect(wrapper).toEqual(null);
    });

    it('snapshot when errors are exist', () => {
        const errors = {
            'Email or password': 'is invalid',
            'Status: 400. Bad request': 'Please, try again'
        };
        const wrapper = renderer.create(<ListErrors errors={errors} />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
})
