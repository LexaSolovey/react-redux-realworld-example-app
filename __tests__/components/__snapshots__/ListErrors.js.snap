// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> ListErrors component snapshot when errors are exist 1`] = `
<ul
  className="error-messages"
>
  <li>
    Email or password
     
    is invalid
  </li>
  <li>
    Status: 400. Bad request
     
    Please, try again
  </li>
</ul>
`;
